src/utransport.rs

```diff
@@ -37,60 +37,28 @@ lazy_static! {
 }
 
 #[inline]
-fn invoke_block_callback(listener: &Arc<dyn UListener>, resp_msg: Result<UMessage, &str>) {
-    match resp_msg {
-        Ok(umsg) => match Handle::try_current() {
-            Ok(handle) => {
-                task::block_in_place(|| {
-                    handle.block_on(listener.on_receive(umsg));
-                });
-            }
-            Err(_) => {
-                TOKIO_RUNTIME
-                    .lock()
-                    .unwrap()
-                    .block_on(listener.on_receive(umsg));
-            }
-        },
-        Err(err_msg) => {
-            error!("{err_msg}");
-            match Handle::try_current() {
-                Ok(handle) => {
-                    task::block_in_place(|| {
-                        handle.block_on(
-                            listener.on_error(UStatus::fail_with_code(UCode::INTERNAL, err_msg)),
-                        );
-                    });
-                }
-                Err(_) => {
-                    TOKIO_RUNTIME.lock().unwrap().block_on(
-                        listener.on_error(UStatus::fail_with_code(UCode::INTERNAL, err_msg)),
-                    );
-                }
-            }
+fn invoke_block_callback(listener: &Arc<dyn UListener>, resp_msg: UMessage) {
+    match Handle::try_current() {
+        Ok(handle) => {
+            task::block_in_place(|| {
+                handle.block_on(listener.on_receive(resp_msg));
+            });
         }
-    }
+        Err(_) => {
+            TOKIO_RUNTIME
```

Comment:
Following back up on this -- did you look into other solutions or is this the way to go about it when the code is running as a part of a different executor (async_std)?

Reply:
Not yet. It's a good topic worth more investigation, but I'll not put it in this PR.
In fact, I should keep the PR clean and do one thing at a time, but I've already put too many things in this PR now, which is not good practice :laughing: